zstd.go

```diff
@@ -46,6 +50,30 @@ func cCompressBound(srcSize int) int {
 	return int(C.ZSTD_compressBound(C.size_t(srcSize)))
 }
 
+// decompressSizeHint tries to give a hint on how much of the output buffer size we should have
+// based on zstd frame descriptors. To prevent DOS from maliciously-created payloads, limit the size
+func decompressSizeHint(src []byte) int {
+	// 1 MB or 10x input size
+	upperBound := 10 * len(src)
+	if upperBound < 1000*1000 {
```

Comment:
nit, should we make this a constant with some documentation on the value?

Reply:
yes, good call! I'm leaning toward making it private though since I don't think it would be useful as a package variage, do you agree ?

Reply:
Sounds good to me

Reply:
added in https://github.com/DataDog/zstd/pull/115/commits/f973148a16db7d475e63b6bcd99b2f0e77034c30